packages/lib/modules/pool/actions/initialize/usePathToInitializePool.ts

```diff
@@ -0,0 +1,249 @@
+      address: poolAddressParam,
+      abi: erc20Abi,
+      chainId,
+      functionName,
+    }))
+
+    const vaultContractReads = vaultFunctionNames.map(functionName => ({
+      address: AddressProvider.Vault(chainId),
+      chainId,
+      abi: vaultExtensionAbi_V3,
+      functionName,
+      args: [poolAddressParam],
+    }))
+
+    return [...poolContractReads, ...vaultContractReads]
+  }, [poolAddressParam, chainId])
+
+  const { data: poolData, isLoading: isLoadingPoolData } = useReadContracts({
+    query: { enabled: poolDataContracts.length > 0 && shouldUsePathToInitialize },
+    contracts: poolDataContracts,
+  })
+
+  const tokenFunctionNames = ['name', 'symbol', 'decimals']
+
+  const tokenContracts = useMemo(() => {
+    if (!poolData) return []
+
+    const tokenAddresses = (poolData[2].result as unknown as any[])[0] as Address[]
+
+    return tokenFunctionNames
+      .flatMap((functionName: string) =>
+        tokenAddresses.map((address: Address) => ({
+          address,
+          abi: erc20Abi,
+          chainId,
+          functionName,
+        }))
+      )
+      .sort((a, b) => a.address.localeCompare(b.address))
+  }, [chainId, poolData])
+
+  const { data: poolTokenDetails, isLoading: isLoadingPoolTokenDetails } = useReadContracts({
+    query: {
+      enabled: tokenContracts.length > 0,
+    },
+    contracts: tokenContracts,
+  })
+
+  const { data: amplificationParameter, isLoading: isLoadingAmplificationParameter } =
+    useReadContract({
+      address: poolAddressParam,
+      abi: parseAbi(['function getAmplificationParameter() view returns (uint256)']),
+      functionName: 'getAmplificationParameter',
+      chainId,
+      query: { enabled: isStablePool && shouldUsePathToInitialize },
+    })
+
+  const { data: normalizedWeights, isLoading: isLoadingWeights } = useReadContract({
+    address: poolAddressParam,
+    abi: parseAbi(['function getNormalizedWeights() view returns (uint256[])']),
+    functionName: 'getNormalizedWeights',
+    chainId,
+    query: { enabled: isWeightedPool && shouldUsePathToInitialize },
+  })
+
+  const { data: reClammConfig, isLoading: isLoadingReClamm } = useReadContract({
+    address: poolAddressParam,
+    abi: parseAbi([
+      'function getReClammPoolImmutableData() external view returns ((address[] tokens, uint256[] decimalScalingFactors, bool tokenAPriceIncludesRate, bool tokenBPriceIncludesRate, uint256 minSwapFeePercentage, uint256 maxSwapFeePercentage, uint256 initialMinPrice, uint256 initialMaxPrice, uint256 initialTargetPrice, uint256 initialDailyPriceShiftExponent, uint256 initialCenterednessMargin, uint256 maxCenterednessMargin, uint256 maxDailyPriceShiftExponent, uint256 maxDailyPriceRatioUpdateRate, uint256 minPriceRatioUpdateDuration, uint256 minPriceRatioDelta, uint256 balanceRatioAndPriceTolerance))',
+    ]),
+    functionName: 'getReClammPoolImmutableData',
+    chainId,
+    query: { enabled: isReClammPool && shouldUsePathToInitialize },
+  })
```

Comment:
all ABI functions seem available from generated?

Reply:
the options for `getNormalizedWeights` are `balancerV2WeightedPoolV4Abi` or `gyroEclpPoolAbi`

the creation UI deploys weighted v3 pools but the function sig looks the same as older ABIs so should be fine?

Reply:
the return type for `getAmplificationParameter` for `balancerV2ComposableStablePoolV5Abi` is different from the v3 return type

Reply:
you can add all contracts you need here: https://github.com/balancer/frontend-monorepo/blob/103a1ea731d5affb639b3db4814a36c7bdff77b2/apps/frontend-v3/wagmi.config.ts#L27
and then run `gen:wagmi` from the app's `package.json`